server/ControlPlane/OpenApi/OpenApi.cs

```diff
@@ -72,6 +83,106 @@ public static void AddOpenApi(this IHostApplicationBuilder builder)
     public static void UseOpenApi(this WebApplication app)
     {
         app.UseSwagger();
+
+        if (app.Environment.IsDevelopment())
+        {
+            app.UseSwaggerUI(options =>
+            {
+                foreach (var description in app.DescribeApiVersions())
+                {
+                    var url = $"/swagger/{description.GroupName}/swagger.json";
+                    var name = description.GroupName.ToUpperInvariant();
+                    options.SwaggerEndpoint(url, name);
+                }
+            });
+        }
+    }
+}
+
+internal sealed class ConfigureSwaggerOptions : IConfigureOptions<SwaggerGenOptions>
+{
+    private readonly IApiVersionDescriptionProvider _provider;
+
+    public ConfigureSwaggerOptions(IApiVersionDescriptionProvider provider) =>
+      _provider = provider;
+
+    public void Configure(SwaggerGenOptions options)
+    {
+        foreach (var description in _provider.ApiVersionDescriptions)
+        {
+            options.SwaggerDoc(
+              description.GroupName,
+                new OpenApiInfo()
+                {
+                    Title = $"Tyger Server",
+                    Version = description.ApiVersion.ToString(),
+                });
+        }
+    }
+}
+
+/// <summary>
+/// Represents the OpenAPI/Swashbuckle operation filter used to document information provided, but not used.
+/// </summary>
+/// <remarks>This <see cref="IOperationFilter"/> is only required due to bugs in the <see cref="SwaggerGenerator"/>.
+/// Once they are fixed and published, this class can be removed.</remarks>
+internal sealed class SwaggerDefaultValuesOperationFilter : IOperationFilter
+{
+    /// <inheritdoc />
+    public void Apply(OpenApiOperation operation, OperationFilterContext context)
+    {
+        var apiDescription = context.ApiDescription;
+
+        operation.Deprecated |= apiDescription.IsDeprecated();
+
+        // REF: https://github.com/domaindrivendev/Swashbuckle.AspNetCore/issues/1752#issue-663991077
+        foreach (var responseType in context.ApiDescription.SupportedResponseTypes)
+        {
+            // REF: https://github.com/domaindrivendev/Swashbuckle.AspNetCore/blob/b7cf75e7905050305b115dd96640ddd6e74c7ac9/src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs#L383-L387
+            var responseKey = responseType.IsDefaultResponse ? "default" : responseType.StatusCode.ToString();
+            var response = operation.Responses[responseKey];
+
+            foreach (var contentType in response.Content.Keys)
+            {
+                if (!responseType.ApiResponseFormats.Any(x => x.MediaType == contentType))
+                {
+                    response.Content.Remove(contentType);
```

Comment:
You'll get an exception in the foreach if you modify the dictionary while enumerating.